transformations/correct_common_misspellings/transformation.py

```diff
@@ -0,0 +1,4117 @@
+import spacy
+
+from initialize import spacy_nlp
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+
+class CorrectCommonMisspellings(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+        TaskType.TEXT_TAGGING,
+    ]
+    languages = ["en"]
+
+    def __init__(self):
+        super().__init__()
+        self.COMMON_MISSPELLINGS_DICT = get_common_misspellings_dict()
+        self.nlp = spacy_nlp if spacy_nlp else spacy.load("en_core_web_sm")
+
+    def generate(self, sentence: str):
+        # print('input:', sentence)
+        doc = self.nlp(sentence)
+        spaces = [True if tok.whitespace_ else False for tok in doc]
+
+        # replace misspelled words
+        perturbed_text = []
+        for index, token in enumerate(doc):
+            s = token.text
+            if s in self.COMMON_MISSPELLINGS_DICT:
+                s = self.COMMON_MISSPELLINGS_DICT[s]
+            perturbed_text.append(s)
```

Comment:
doesn't matter, but:
`perturbed_text = [ self.COMMON_MISSPELLINGS_DICT[s.text] for s in doc if s.text in self.COMMON_MISSPELLINGS_DICT]`

Is likely easier to parse

Reply:
I'd actually just consider replacing the whole function with

```
perturbed_text = [ self.COMMON_MISSPELLINGS_DICT.get(token.text, token.text) + " " for token in doc 
                              if token.whitespace_ 
                              else self.COMMON_MISSPELLINGS_DICT.get(token.text, token.text) ]                      
```

Reply:
changed